Update how location details are displayed across app

mforner13 opened the pull request:
Due to the recent updates to the `Location` model (#13), which meant that masterclasses could either be remote or have a location, the way masterclasses are displayed in templates is no longer correct.

This PR updates the `index` and `masterclass_profile` templates to:
1) Display the location as `Remote` if `masterclass.is_remote` is True
2) Only display the joining URL and instructions if a user is an attendee of that masterclass

I took the latter decision to avoid more people than the instructor has set as `max_attendees` being able to join the masterclass. 

I have added tests for the following cases:

- Masterclass is remote
- Masterclass has a location
- Masterclass is remote and user isn't an attendee
- Masterclass is remote and user is an attendee

TODO

- Create migration for removing old fields from `Location` model now that they are no longer used in templates

mforner13 committed:
Display materclasses on homepage if draft = False

mforner13 committed:
Correct redirect for signup confirmation

mforner13 committed:
Display 'remote' as location when applicable

mforner13 committed:
Show remote joining info when user is attendee

mforner13 committed:
Represent Location instances with name

mforner13 committed:
Test correct location data is displayed in profile

Add tests to make sure the correct location details are displayed in a masterclass profile depending on if it is remote or has a location. If remote, test that the joining instructions aren't displayed unless a user is an attendee.

mforner13 commented on tests/test_masterclasses.py:
```diff
@@ -82,12 +83,24 @@ def new_masterclass_content_data_category(db, blank_session):
+
+@pytest.fixture()
+def new_masterclass_with_location(db, blank_session):
+    m = Masterclass(id=1, is_remote=False, max_attendees=10, timestamp=datetime(2020, 10, 30, 15, 30), location_id=1)
```
The `masterclass_profile` template throws an error if `max_attendees` and `timestamp` are not set.

jonodrew replied:
Interesting - are users forced to complete those when they create a `Masterclass`?

mforner13 replied:
Yep they will be

mforner13 commented on app/routes.py:
```diff
@@ -9,7 +9,7 @@
 @login_required
 def index():
-    masterclasses = Masterclass.query.filter_by(draft=None).order_by(Masterclass.timestamp.asc()).all()
+    masterclasses = Masterclass.query.filter_by(draft=False).order_by(Masterclass.timestamp.asc()).all()
```
I think it makes more sense for `draft` to be set as False when not true, rather than None.

jonodrew replied:
Probably fair - but this changes the search. So why were we searching for `None` before?

mforner13 replied:
Basically before I wiped my local db all of the existing masterclasses just had `None` for the draft field. However after wiping them, I can add them as `False`, as I will when a user comes to the end of the create masterclass flow.

jonodrew commented on app/templates/index.html:
```diff
@@ -12,7 +12,15 @@ <h2 class="govuk-heading-m"><a class="govuk-link--no-visited-state" href="{{ url
+                    {% else %}
+                        <td class="govuk-body">
+                            {{ masterclass.location.name }}, {{ masterclass.location.address }}
+                        </td>
```
This could be simpler, I think? Since the content has to go inside a `td class="govuk-body"`, could you put the if/else inside that class?

mforner13 replied:
Good point - have updated :)

jonodrew commented on tests/test_masterclasses.py:
```diff
@@ -164,18 +177,38 @@ def test_draft_masterclass_id_is_added_to_session(logged_in_user, blank_session)
+    logged_in_user.post('/masterclass/2')
+    response = logged_in_user.get('/masterclass/2')
+    assert '<h2 class="govuk-heading-m">Joining link</h2>' in response.get_data(as_text=True)
+    # This not working
```
Are you committing a non-working test?

mforner13 replied:
Ah I had forgotten to take that out once I got it working again. Have now removed.

jonodrew reviewed (commented):
A few things I'm unsure of here and I'd like to see your views on please!

mforner13 committed:
Rename variable for masterclass passed to template

For simplicity and consistency with masterclass instances passed to other templates, I have renamed masterclass_data just masterclass.

jonodrew reviewed (approved):
Looks great to me!
